primitives/src/types/jobs.rs

```diff
@@ -55,17 +55,101 @@ pub enum JobType<AccountId> {
 	DKG(DKGJobType<AccountId>),
 	/// DKG Signature job type.
 	DKGSignature(DKGSignatureJobType),
-	/// (zk-SNARK) Phase One job type.
-	ZkSaasPhaseOne(ZkSaasPhaseOneJobType<AccountId>),
-	/// (zk-SNARK) Phase Two job type.
-	ZkSaasPhaseTwo(ZkSaasPhaseTwoJobType),
+	/// (zk-SNARK) Create Circuit job type.
+	ZkSaasCircuit(ZkSaasCircuitJobType<AccountId>),
+	/// (zk-SNARK) Create Proof job type.
+	ZkSaasProve(ZkSaasProveJobType),
 }
 
+/// Enum representing different types of data sources.
+#[derive(PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, Clone)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum HyperData {
+	/// Raw data, stored on-chain.
+	///
+	/// Only use this for small files.
+	Raw(Vec<u8>),
```

Comment:
Do these need to be bounded Vecs?

Reply:
cc @1xstj Here, maybe we can update this later, since it will require another generic parameter